File: lab4/app/src/main/java/com/example/lab4/MyStoryAdapter.java
package com.example.lab4;

import android.content.Context;
import android.content.Intent;
import android.media.Image;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class MyStoryAdapter extends RecyclerView.Adapter<MyStoryAdapter.MyViewHolder>{

    String storyName[];
    int storyPic[];
    Context context;

    public MyStoryAdapter(Context context, String storyName[], int storyPic[]){
        this.context = context;
        this.storyName = storyName;
        this.storyPic = storyPic;
    }

    @NonNull
    @Override
    public MyStoryAdapter.MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from((context));
        View view = inflater.inflate(R.layout.stories, parent, false);
        return new MyStoryAdapter.MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MyStoryAdapter.MyViewHolder holder, final int position) {
        holder.storyPic.setImageResource(storyPic[position]);
        holder.storyName.setText(storyName[position]);
    }

    public int getItemCount() {
        return storyPic.length;
    }

    public class MyViewHolder extends RecyclerView.ViewHolder{

        TextView storyName;
        ImageView storyPic;

        public MyViewHolder(@NonNull View itemView){
            super(itemView);
            storyName = itemView.findViewById(R.id.storyName);
            storyPic = itemView.findViewById(R.id.storyPic);
        }

    }


}

File: inst_na_kotline/app/src/main/java/com/example/inst_na_kotline/MyStoryAdapter.kt
package com.example.inst_na_kotline

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView


class MyStoryAdapter(
    var context: Context,
    var storyName: Array<String>,
    var storyPic: IntArray
) :
    RecyclerView.Adapter<MyStoryAdapter.MyViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val inflater = LayoutInflater.from(context)
        val view = inflater.inflate(R.layout.stories, parent, false)
        return MyViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.storyPic.setImageResource(storyPic[position])
        holder.storyName.text = storyName[position]
    }

    override fun getItemCount(): Int {
        return storyPic.size
    }

    inner class MyViewHolder(itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        var storyName: TextView
        var storyPic: ImageView

        init {
            storyName = itemView.findViewById(R.id.storyName)
            storyPic = itemView.findViewById(R.id.storyPic)
        }
    }

}
